lib/msal-node/src/client/PublicClientApplication.ts

```diff
@@ -74,4 +84,83 @@ export class PublicClientApplication extends ClientApplication implements IPubli
             throw e;
         }
     }
+
+    /**
+     * Acquires a token by requesting an Authorization code then exchanging it for a token.
+     */
+    async acquireTokenInteractive(request: InteractiveRequest): Promise<AuthenticationResult> {
+        const { verifier, challenge } = await this.cryptoProvider.generatePkceCodes();
+        const { openBrowser, successTemplate, errorTemplate, ...remainingProperties } = request;
+
+        return new Promise(async (resolve, reject) => {
+            const server = createServer((req: IncomingMessage, res: ServerResponse) => {
```

Comment:
I think it would be cleaner if you created a `LoopbackClient` or somthing that hides most the implementation details of how this work.

Reply:
Done